src/dataflow/src/source/kafka.rs

```diff
@@ -801,6 +810,15 @@ impl ConsumerContext for GlueConsumerContext {
     }
 }
 
-fn has_partition(worker_id: i32, worker_count: i32, partition_id: i32) -> bool {
-    (partition_id % worker_count) == worker_id
+fn has_partition(
+    source_id: GlobalId,
+    worker_id: i32,
+    worker_count: i32,
+    partition_id: i32,
+) -> bool {
+    assert!(worker_id >= 0);
+    assert!(worker_count > worker_id);
+    assert!(partition_id >= 0);
+    let hash = source_id.hashed() + partition_id as u64;
```

Comment:
Careful about overflow here - I think you should mod before the addition or do the addition as u128

Reply:
hmmm if I mod before the addition then I think I have to do something like `(source_id.hashed() % worker_count as u64 + partition_id as u64 % worker_count as u64)` because of the way mod distributes over addition

Reply:
what i really just want in this case I think is 32 bits of randomness from `hashed` instead of 64

Reply:
ok changed the code to now only retain the top 32 bits from `hashed()` -- I sure hope that this hash function _waves hands_ has the same amount of randomness in its top bits as its bottom bits

Reply:
Yeah, since worker_count and partition_id are i32s you can get away with two mods instead of three:

```
let worker_count = worker_count as u64;
let hash = (source_id.hashed() % worker_count) + partition_id as u64;
(hash % worker_count) == worker_id as u64;
```

To me that's preferable to reasoning about the randomness of different bit positions in the hash function, but entirely your call. 🙂

Reply:
ugh was lead astray by this description of how mod distributes by wikipedia https://en.wikipedia.org/wiki/Modulo_operation and thought that `((a mod k) + b) mod k)` wouldn't be equivalent to `(a + b) mod k` -- thats false. yeah your way is much better

Reply:
although I will say that doing this involves the same reasoning right? like if we take `hashed % 20` we are hoping that the bottom 5 bits are just as random as the rest of the number?

Reply:
> although I will say that doing this involves the same reasoning right? 

Hah! You're right - I rescind my mod over shift suggestion.